tendermint/src/types/proto/validator.rs

```diff
@@ -1,22 +1,62 @@
-use address::{AddressError, ValAddress};
+use address::ValAddress;
 use serde::{Deserialize, Serialize};
-use ux::u63;
 
 use crate::error::Error;
 
 use super::crypto::PublicKey;
 
+const MAX_VALIDATOR_POWER: u64 = 8198552921648689607;
+/// VotingPower holds validator power and guarantees that its value is less than
+/// 8198552921648689607.
+// TODO: can be a Copy type
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+pub struct VotingPower {
+    // look at https://github.com/tendermint/tendermint/issues/2985
+    // https://github.com/tendermint/tendermint/issues/7913
+    power: u64,
+}
+
+impl VotingPower {
+    pub fn new(power: u64) -> Result<VotingPower, Error> {
+        if power > MAX_VALIDATOR_POWER {
+            return Err(Error::InvalidData(format!(
+                "validator power is greater than max validator power {MAX_VALIDATOR_POWER}"
+            )));
+        }
+
+        Ok(VotingPower { power })
+    }
+
+    pub fn power(&self) -> u64 {
+        self.power
+    }
+}
+
+impl TryFrom<u64> for VotingPower {
+    type Error = Error;
+
+    fn try_from(power: u64) -> Result<Self, Self::Error> {
+        Self::new(power)
+    }
+}
+
+impl From<VotingPower> for u64 {
+    fn from(power: VotingPower) -> Self {
+        power.power()
+    }
+}
+
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
 pub struct ValidatorUpdate {
     pub pub_key: PublicKey,
-    pub power: i64,
+    pub power: VotingPower,
 }
 
 impl From<ValidatorUpdate> for inner::ValidatorUpdate {
     fn from(ValidatorUpdate { pub_key, power }: ValidatorUpdate) -> Self {
         Self {
             pub_key: Some(pub_key.into()),
-            power,
+            power: power.power() as i64,
```

Comment:
Even though this is guaranteed to be safe (due to the guarantees on the `VotingPower` max value, it might be cleaner to have a `From<VotingPower` implementation for `i64`.